### PR TITLE
Add support for creating a horizontally-tiling MenuBar

### PR DESCRIPTION
**Pull request by smoogipooo**
Prereqs:
- [x] https://github.com/ppy/osu-framework/pull/1007

Follows the example built in the PR above to create a bar that contains a collection of menus.

### CODE REVIEW

File: osu.Framework/Graphics/UserInterface/MenuBar.cs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            });
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="MenuItem"/>s contained by this <see cref="MenuBar"/>.
+        /// </summary>
+        public IReadOnlyList<MenuItem> Items
+        {
+            get { return ItemsContainer.Select(i => i.Item).ToList(); }
+            set { value?.ForEach(Add); }
```
Comment (peppy) [inline osu.Framework/Graphics/UserInterface/MenuBar.cs]: i don't like allowing null setting here. the return would be an empty list when the setter is setting null, which goes against our policy of get-set not warping the data.

File: osu.Framework/Graphics/UserInterface/Menu.cs [hunk 2]
```diff
@@ -39,8 +40,7 @@ public IReadOnlyList<MenuItem> Items
             {
                 itemsContainer.Clear();
 
-                foreach (var item in value)
-                    Add(item);
+                value?.ForEach(Add);
```
Comment (peppy) [inline osu.Framework/Graphics/UserInterface/Menu.cs]: same here

File: osu.Framework/Graphics/UserInterface/MenuBar.cs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+        /// <summary>
+        /// Clears all <see cref="MenuItem"/>s from this <see cref="MenuBar"/>.
+        /// </summary>
+        public void Clear() => ItemsContainer.Clear();
+
+        protected override bool OnMouseMove(InputState state)
+        {
+            // Find the previously-opened item
+            var previousItem = ItemsContainer.FirstOrDefault(i => i.State == MenuState.Opened);
```
Comment (peppy) [inline osu.Framework/Graphics/UserInterface/MenuBar.cs]: why not just store a `CurrentMenu`, check if it's null or still hovered, then (and only then) to an iteration over children?

File: osu.Framework/Graphics/UserInterface/MenuBar.cs (hunk 1 above)
Comment (smoogipooo) [reply osu.Framework/Graphics/UserInterface/MenuBar.cs]: You would rather this was blocked in https://github.com/ppy/osu-framework/pull/1009/files/6617f3c0ad4e2f7a40f8c86b2bfbeedf56ad453f#diff-a735c980f549eb539827ac1fcedbc4aeR132 ?

File: osu.Framework/Graphics/UserInterface/MenuBar.cs (hunk 3 above)
Comment (smoogipooo) [reply osu.Framework/Graphics/UserInterface/MenuBar.cs]: Store it by callbacking clicks from children?

File: osu.Framework/Graphics/UserInterface/MenuBar.cs (hunk 1 above)
Comment (peppy) [reply osu.Framework/Graphics/UserInterface/MenuBar.cs]: definitely, unless you make the default an empty array for child items.

File: osu.Framework/Graphics/UserInterface/MenuBar.cs (hunk 3 above)
Comment (smoogipooo) [reply osu.Framework/Graphics/UserInterface/MenuBar.cs]: Clicks and click-ups? Eh. Got any reasoning for going for that solution vs. what's in here currently?

Comment (peppy) [reply osu.Framework/Graphics/UserInterface/MenuBar.cs]: Yeah, I'd make a `StateChanged` method which we bind to.

Comment (peppy) [reply osu.Framework/Graphics/UserInterface/MenuBar.cs]: is a lot cleaner. ensures if a menu item is closed, the `StateChanged` callback will make sure the parent menu is aware of the state change

**Comment by smoogipooo** [comment]
Closing for rewrite as discussed.

**Pull request closed without merging**